model/riscv_types.sail

```diff
@@ -128,6 +128,23 @@ function arch_to_bits(a : Architecture) -> arch_xlen =
     RV128 => 0b11
   }
 
+
+/* model-internal exceptions */
+
+union exception = {
+  Error_not_implemented : string,
+  Error_internal_error  : unit
+}
+
+val not_implemented : forall ('a : Type). string -> 'a effect {escape}
+function not_implemented message = throw(Error_not_implemented(message))
+
+val internal_error : forall ('a : Type). string -> 'a effect {escape}
+function internal_error(s) = {
+    assert (false, s);
```

Comment:
I am not very famillar with SAIL assert implementation, would this print call stack ? otherwise it might be helpfull to print  __FILE__ and __LINE__  as requested in this issue https://github.com/rems-project/sail/issues/175 ?

Reply:
No, unfortunately it will just print a message and `exit(1)`. You'd need a debugger to get a stack trace.

Reply:
__FILE__ and __LINE__ would have to be passed to this function as arguments to be useful. I think Sail might have equivalent macros but all calls to `internal_error` would need updated.

Reply:
Given that the `internal_error` function is already in use and cannot usefully display the file and line number or stack trace can we just ensure that the assert messages are sufficiently different and detailed?

Reply:
There is only a small number of calls to internal_error (about 40), so it would be easy to update. Right now, some of them have identical error messages (eg. "invalid physical address in TLB"). That's suboptimal.